server/app/routes/orders/index.js

```diff
@@ -91,29 +81,65 @@ router.put('/:orderId', function (req, res, next) {
 router.post('/', function (req, res, next) {
 	// security consideration:
 	// what if someone spams us with orders via postman?
+	var shippingAddress = req.body.shippingAddress,
+		shippingEmail = req.body.shippingEmail;
 	
-
+	if (!shippingEmail || !shippingAddress) return next({status: 400, message: "Shipping address and email are required"});
 
 	var newOrder = {
-		//instead of getting the cart sent from the frontend
-		// we could also just use the one on the user or session
-		purchasedItems : req.body.cart, 
-		shippingAddress : req.body.shippingAddress,
-		shippingEmail : req.body.shippingEmail
+		purchasedItems: req.session.cart,
+		shippingAddress : shippingAddress,
+		shippingEmail : shippingEmail
 	};
-	// If a guest creates an order there won't be a user property on the order
-	if(req.user) newOrder.user = user._id;
-
-	Order.create(newOrder)
-	.then(function(createdOrder) {
-		// delete cart from the session TODO!!!
-		res.status(201).send(createdOrder);
-	})
-	// This should validation errors: Empty carts etc.
-	// so the frontend will receive an error
-	.then(null, function(err) {
-		next({status: 400, message: "Validation Error: Could not create order"});
-	});
+	
+	//if there is a logged in user, the cart should be on the user schema
+	if (req.user) {
+		User.findById(req.user._id)
+			.populate('cart.product')
```

Comment:
We should populate this on the cart page that way the request will always have the complete purchasedItem array.